### ISSUE TITLE
VK_NULL_HANDLE MSVC Compilation error with v1.2.175 headers (vulkan_core.h)

### ISSUE THREAD

**Issue by ifarbod** [KhronosGroup/Vulkan-Docs#1502]
I'm using `VK_NULL_HANDLE` for almost all Vulkan structure pointers but as of updating to the latest headers, I can't compile my app on MSVC anymore.

For example, for the code below:

```cpp
VkRenderpass renderpass = VK_NULL_HANDLE;
```

I'm getting an error where it says it can't initialize the variable with an incompatible type (`void*`).
This is due to MSVC's `__cplusplus` macro being stuck on C++98 for decades. A correct fix for this would be updating the corresponding macro check like this:

```cpp
        //#if __cplusplus >= 201103L
        #if __cplusplus >= 201103L || (defined(_MSVC_LANG) && (_MSVC_LANG >= 201103L))
            #define VK_NULL_HANDLE nullptr
        #else
            #define VK_NULL_HANDLE ((void*)0)
        #endif
```

**Comment by polarina** [comment]
The `VK_NULL_HANDLE` is for non-dispatchable handles only; `VkInstance` is a dispatchable handle. You can use either `NULL` or `nullptr` for dispatchable handles.

**Comment by ifarbod** [comment]
@polarina Using `VkInstance` was merely an example actually, I've edited it but my point still stands. Many codebases rely on `VK_NULL_HANDLE` and this is a somewhat breaking change (without the proposed fix), not just mine.

**Comment by oddhack** [comment]
Thanks. We will push out an update to Vulkan-Headers for the MSVC issue shortly, but recommend falling back to the previous headers (or the headers that came with your SDK) in the meantime.

**Comment by mark-lunarg** [comment]
And a  [possibly-related clang issue](https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/2707)?

**Comment by oddhack** [comment]
@ifarbod (and anyone else affected), could you test the attached header (which should be identical to your proposed fix) and report back?
 
 [vulkan_core.zip](https://github.com/KhronosGroup/Vulkan-Docs/files/6321779/vulkan_core.zip)

**Comment by ifarbod** [comment]
@oddhack Yes, it works just fine. That's identical to what I've done in my project. I hope the fix gets upstream so I don't have to manually patch the header for each release.

**Comment by oddhack** [comment]
> @oddhack Yes, it works just fine. That's identical to what I've done in my project. I hope the fix gets upstream so I don't have to manually patch the header for each release.

It should be going out on Monday or Tuesday at latest.

**Comment by oddhack** [comment]
This should be fixed in the 1.2.176 update.
